include/JSStringProxy.hh

```diff
@@ -0,0 +1,33 @@
+/**
+ * @file JSStringProxy.hh
+ * @author Caleb Aikens (caleb@distributive.network)
+ * @brief JSStringProxy is a custom C-implemented python type that derives from str. It acts as a proxy for JSStrings from Spidermonkey, and behaves like a str would.
+ * @version 0.1
+ * @date 2024-01-03
+ *
+ * Copyright (c) 2024 Distributive Corp.
+ *
+ */
+
+#ifndef PythonMonkey_JSStringProxy_
+#define PythonMonkey_JSStringProxy_
+
+#include <jsapi.h>
+
+#include <Python.h>
+
+/**
+ * @brief The typedef for the backing store that will be used by JSStringProxy objects. All it contains is a pointer to the JSString
+ *
+ */
+typedef struct {
+  PyUnicodeObject str;
+  JS::PersistentRootedValue jsString;
```

Comment:
why not an object as above?

Reply:
strings are not objects

Reply:
I mean the type, JS::PersistentRootedValue vs JS::PersistentRootedObject

Reply:
JSStrings are not Objects, you cannot store a JSString in a JS::PersistentRootedObject unless it is a boxed string, but I am using primitive strings here.